hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/table/HoodieFlinkMergeOnReadTable.java

```diff
@@ -117,7 +119,7 @@ public HoodieWriteMetadata<List<WriteStatus>> compact(
       HoodieEngineContext context, String compactionInstantTime) {
     RunCompactionActionExecutor compactionExecutor = new RunCompactionActionExecutor(
         context, config, this, compactionInstantTime, new HoodieFlinkMergeOnReadTableCompactor(),
-        new HoodieFlinkCopyOnWriteTable(config, context, getMetaClient()), WriteOperationType.COMPACT);
+        this, WriteOperationType.COMPACT);
```

Comment:
sane. but is something working today, since the new table object inits everything again all over? - that may break

Reply:
The compaction flow creates the table object right before triggering the action so it should be fine. The compaction mainly operates off of the plan and does not need to lookup the fsview or anything like that again.